### PR TITLE
add sphere union spec

### PR DESCRIPTION
**Pull request by jglaser**
## Description

We should support sphere unions in gsd, for downstream visualization in ovito.

## Motivation and Context

Visualize more complex shapes, such as proteins

## How Has This Been Tested?

Built documentation, looks ok.

## Change log

```
Add shape spec for sphere unions
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/gsd/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**GSD Contributor Agreement**](https://github.com/glotzerlab/gsd/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/gsd/blob/master/doc/credits.rst).

### CODE REVIEW

**Review by joaander — APPROVED** [review]
Thanks! The shape specification looks good.

File: doc/shapes.rst [hunk 1]
```diff
@@ -137,3 +137,26 @@ Example::
...
+=============== =============== ===== ==== ======= ======
+radii           Sphere radii    float Nx1          length
+centers         Sphere centers  float Nx3          length
+=============== =============== ===== ==== ======= ======
+
+Example::
+
+    {
+        "type": "SphereUnion",
+        "centers": [[0, 0, 1.0], [0, 1.0, 0], [1.0, 0, 0]]
```
Comment (bdice) [inline doc/shapes.rst]: Missing comma at end of line.

**Review by bdice — COMMENTED** [review]
@jglaser One minor edit requested. Otherwise good!

**Comment by bdice** [comment]
@jglaser Also, a major change request: we use **diameters, not radii**, in both HOOMD-blue and garnett.

https://hoomd-blue.readthedocs.io/en/stable/module-hpmc-integrate.html#hoomd.hpmc.integrate.sphere_union

https://garnett.readthedocs.io/en/stable/shapes.html#garnett.shapes.SphereUnionShape

**Comment by jglaser** [comment]
OK. HOOMD uses radii *internally*, but I see now that the API for a sphere in the shape spec uses the diameter. So yes, we can make that consistent.

**Pull request merged**